main.tf

```diff
@@ -407,24 +407,34 @@ locals {
   })
 }
 
-resource "aws_launch_configuration" "lc" {
+resource "aws_launch_template" "lt" {
```

Comment:
One suggestion would be to add `tags = local.tags` in here so that the launch template resource has tagging added.